.github/workflows/coverage.yml

```diff
@@ -32,13 +32,12 @@ jobs:
         with:
           # Fetch depth 0 required to compare against `main`
           fetch-depth: 0
-
-      # Load a specific version of Python
-      - name: Setup python
-        uses: actions/setup-python@v4
+      - uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python }}
           architecture: x64
+      - uses: syphar/restore-virtualenv@v1
+        id: cache-virtualenv
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏Can you add a comment for what this action is for/why it's needed? Same for everywhere it's used.